From ecaa29efa5a3bf74abcb36581eb8d953363ee2ba Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Fri, 12 Sep 2014 01:56:56 -0400
Subject: [drm] dp-mst-helper: Avoid reading uninitialized value

Message-id: <1410487019-24661-19-git-send-email-airlied@redhat.com>
Patchwork-id: 93927
O-Subject: [RHEL7.1 BZ1140440 PATCH 18/21] drm/dp-mst-helper: Avoid reading uninitialized value
Bugzilla: 1140440
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Rob Clark <rclark@redhat.com>

From: Damien Lespiau <damien.lespiau@intel.com>

bugzilla: 1140440
upstream: 7389ad4b6515c2de6402bfafdfebf0b319790d16

A static analysis tool tells me that we could try to read an
uninitialized 'ret' value. Plug that.

Signed-off-by: Damien Lespiau <damien.lespiau@intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Signed-off-by: Dave Airlie <airlied@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/drm_dp_mst_topology.c b/drivers/gpu/drm/drm_dp_mst_topology.c
index 97e3922..fbe7f04 100644
--- a/drivers/gpu/drm/drm_dp_mst_topology.c
+++ b/drivers/gpu/drm/drm_dp_mst_topology.c
@@ -1646,7 +1646,7 @@ int drm_dp_update_payload_part2(struct drm_dp_mst_topology_mgr *mgr)
 {
  struct drm_dp_mst_port *port;
  int i;
- int ret;
+ int ret = 0;
  mutex_lock(&mgr->payload_lock);
  for (i = 0; i < mgr->max_payloads; i++) {
 
-- 
1.7.1